tests/Utils/ArrayLogger.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\Test\Utils;
+
+use SimpleSAML\Configuration;
+use SimpleSAML\Logger\LoggingHandlerInterface;
+
+class ArrayLogger
+    implements LoggingHandlerInterface
+{
+    /**
+     * @var array List of log entries by level
+     */
+    public $logs = [];
```

Comment:
Since master has PHP 7.4 as minimum, use typed propery instead of phpdoc annotation?